### PR TITLE
Protect dnsdist client and query rings with a RW lock

### PR DESCRIPTION
**Pull request by rgacogne**

### CODE REVIEW

File: pdns/dnsdistdist/Makefile.am [hunk 1]
```diff
@@ -59,7 +60,7 @@ dnsdist_SOURCES = \
 
 dnsdist_LDFLAGS = \
 	$(AM_LDFLAGS) \
-	-pthread 
+	-pthread
```
Comment (cmouse) [inline pdns/dnsdistdist/Makefile.am]: whitespace nit

File: pdns/dnsdistdist/Makefile.am [hunk 2]
```diff
@@ -67,4 +68,3 @@ dnsdist_LDADD = \
 	$(RT_LIBS) \
 	$(YAHTTP_LIBS) \
 	$(LIBSODIUM_LIBS)
-
```
Comment (cmouse) [inline pdns/dnsdistdist/Makefile.am]: whitespace nit

**Comment by rgacogne** [comment]
fixed the removed trailing whitespaces rightly reported by @cmouse

**Comment by rgacogne** [comment]
rebased

**Comment by ahupowerdns** [comment]
ok - are you sure IDState needs a lock? I tried very hard to make it not need one!

**Pull request merged**